addons/mail/static/src/js/chatter.js

```diff
@@ -160,6 +160,10 @@ var Chatter = Widget.extend({
             self._updateMentionSuggestions();
         });
         this._updateAttachmentCounter();
+
+        if (this.record.data.message_attachment_count || 0 > this.attachments.length) {
+            this._fetchAttachments();
```

Comment:
```js
if (!this._areAttachmentsLoaded) {
   def = this._fetchAttachments();
}
```

Reply:
I changed the `_updateAttachmentCounter` method in order to reset the flag when the counter changes